config/jobs/kubernetes/kops/build_jobs.py

```diff
@@ -752,6 +752,7 @@ def generate_distros():
                        kops_channel='alpha',
                        name_override=f"kops-aws-distro-image{distro}",
                        extra_dashboards=['kops-distros'],
+                       extra_flags=['--set=spec.assets.containerProxy=registry-sandbox.k8s.io'],
```

Comment:
This will require networking='kubenet' to work, as any Calico assets will b missing from `sandbox.k8s.io`, right?

Reply:
The only assets concerned are the kubernetes container images from `k8s.gcr.io`. PR updated.

Reply:
Yes, but this setting affects all assets, not just the `k8s.gcr.io ones`. Thanks for updating.

Reply:
@hakman What's the best approach if I want to extend this change to other periodic jobs ? Is this change enough or do I need to make changes per use ?

Reply:
Most jobs use CNIs (as kubenet style networking is not really a production use case). This means that setting this proxy will not work for that case, unless the proxy is made to proxy CNI repos too.
Most likely path is to decide on a final dns name for the CDN cached registry and switch all kOps manifests to use it during the kOps 1.24 development cycle. We can decide later during that cycle if we want to keep it or revert to `k8s.gcr.io`.

Reply:
Thanks for the help!

Reply:
My pleasure. Thanks for trying to improve this. I think it will become a huge improvement for many k8s users in the future if successful.